Automattic/wp-calypso
# Framework: use a single localforage config

This PR moves localforage defaults out of config and into `lib/localforage`. The sync handler and redux persistence now share the same browser storage. There is little chance for key collision because the reserved key for redux persistence is 'redux-state' and sync handler keys are hashed.

![store](https://cloud.githubusercontent.com/assets/1270189/13024835/2fe731d0-d1b0-11e5-9f8c-76bbc0a99680.png)

This fixes #3282. 

## Testing instructions

- In `development.json` set `persist-redux` to true
- Start Calypso
- Set localStorage.debug to 'calypso:state'
- Navigate to the posts page
- Examine local browser storage, you should see something similar to the screenshot above.
- Sync handler and the persisted redux store should behave normally and not throw errors.

cc @rralian @retrofox
